src/incubator/TextField/CharCounter.tsx

```diff
@@ -7,12 +7,15 @@ import {CharCounterProps} from './types';
 
 const CharCounter = ({maxLength, charCounterStyle, testID}: CharCounterProps) => {
   const {value} = useContext(FieldContext);
+
+  const textStyle = useMemo(() => [styles.container, charCounterStyle], [charCounterStyle]);
```

Comment:
This is a redundant memoization. when value changes, the char counter always changes and re-redner, so u might as well pass a new style. 

No reason to memoize stuff that don't save re-redners. memoize comes in a price of app memory